### PR TITLE
Update broken OmegaConfigLoader test 

### PR DESCRIPTION
**Pull request by AhdraMeraliQB**
## Description

In #3587 it was pointed out that the test `test_overlapping_patterns` contained an assert statement that was passing trivially. Correcting this assert showed that the logic being tested was redundant, a remnant from the previous [ConfigLoader](https://github.com/kedro-org/kedro/blob/3529c29234ba86a9d3605e3a80d030518f06b311/kedro/config/config.py) setup that was not carried over to the OmegaConfigLoader. As a result, this assert statement has been removed.

## Development notes

The test coverage is unaffected by this removal.

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [x] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

File: tests/config/test_omegaconf_config.py [hunk 1]
```diff
@@ -439,10 +437,6 @@ def test_overlapping_patterns(self, tmp_path, mocker):
         }
         assert catalog == expected_catalog
 
-        # Assert any specific config file was only loaded once
-        expected_path = (tmp_path / "dev" / "user1" / "catalog2.yml").resolve()
-        load_spy.assert_called_once_with(expected_path)
-
```
Comment (noklam) [inline tests/config/test_omegaconf_config.py:445]: Why is this removed?

**Review by merelcht — APPROVED** [review]

**Review by noklam — APPROVED** [review]
